prow/plugins/approve/approvers/approvers_test.go

```diff
@@ -908,3 +908,48 @@ Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a commen
 		t.Errorf("GetMessage() = %+v, want = %+v", *got, want)
 	}
 }
+
+func TestGetMessageMDOwners(t *testing.T) {
+	ap := NewApprovers(
+		Owners{
+			filenames: []string{"a/a.go", "b/README.md"},
+			repo: createFakeRepo(map[string]sets.String{
+				"a":           sets.NewString("Alice"),
+				"b":           sets.NewString("Bill"),
+				"b/README.md": sets.NewString("Doctor"),
+			}),
+			log: logrus.WithField("plugin", "some_plugin"),
+		},
+	)
+	ap.AddAuthorSelfApprover("John", "REFERENCE", false)
+	ap.RequireIssue = true
+	want := `[APPROVALNOTIFIER] This PR is **NOT APPROVED**
+
+This pull-request has been approved by: *<a href="REFERENCE" title="Author self-approved">John</a>*
+To fully approve this pull request, please assign additional approvers.
+We suggest the following additional approvers: **alice**, **doctor**
+
+Assign the PR to them by writing ` + "`/assign @alice @doctor`" + ` in a comment when ready.
+
+*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with ` + "`/approve no-issue`" + `
+
+The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).
+
+The pull request process is described [here](https://git.k8s.io/community/contributors/devel/owners.md#the-code-review-process)
+
+<details open>
+Needs approval from an approver in each of these OWNERS Files:
+
+- **[a/OWNERS](https://github.com/org/project/blob/master/a/OWNERS)**
+- **[b/README.md](https://github.com/org/project/blob/master/b/README.md)**
```

Comment:
I don't read the logic of finding owner header of `.md` files. I wonder if we still try to find the owner header of the md file and list the md file here if some md files don't have an owner header. I notice there are md files in k/website don't have such headers. @cjwagner

Reply:
That isn't a concern here. `.md` files will only be reference in the `RepoOwners` struct if they were found to have valid OWNERS headers (and the headers were successfully parsed) while walking the repo to load the OWNERS info.

Reply:
I modified a test to show that `.md` files without OWNERS headers work as like any other file.